### ISSUE TITLE
build/all.py A required dependency is missing: C:\Program Files\nodejs\node.EXE

### ISSUE THREAD

**Issue by WINOFFRG** [shaka-project/shaka-player#4937]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
4.3.3

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `main`?**
Yes

**What did you do?**

I ran python3 `build/all.py` and got error `[ERROR] *** A required dependency is missing: C:\Program Files\nodejs\node.EXE`

**What did you expect to happen?**
Should work in case of long paths beecause I have cloned shaka player in a monorepo project and want to use it as a dependency, Now I cant move the complete project in some other location as `node_modules` are pretty heavy in total for multiple packages.

**What actually happened?**

I checked for similar issues here and found that this is still open https://github.com/shaka-project/shaka-player/issues/3551 Creating another issue to propose a solution. After some debugging found that we should first fix the error messages on exit as it is confusing. The actual error `[WinError 206] The filename or extension is too long` instead of `[ERROR] *** A required dependency is missing: C:\Program Files\nodejs\node.EXE` https://github.com/shaka-project/shaka-player/blob/38cfc230483133862cc41153931f4dc170eb4a75/build/shakaBuildHelpers.py#L164-L166

Here we try to eslint all the files by providing a config file. The issue is since there are so many files to lint, All the paths add up and node isn't capable of reading args with such a long length. Since path can vary on every machine, Can we lint iteratively?

We can take like 5, 10 or a specific number of files at a time and either spawn all of them async or sync. The advantage of loading them async in batches is 
1. Prevent long paths error from coming up.
2. Decrease the Linting Time.

And a possible case not sure if its advantage or disadvantage, If there are multiple errors they would all be thrown at same time. Not sure about python but in JS we have `Promise.all` and `Promise.allSettled` where we can control if to terminate all the Promises at once in case of an error or not.  If the proposed solution seems good, Kindly assign it to me, I will start working on it when I get time.

**Comment by joeyparrish** [comment]
Please feel free to send PRs that improve error messages and reduce command-line lengths.

However, please do not lint files in batches, as this would provide incomplete feedback (if done serially/sync) or an unreadable mix of outputs being written simultaneously to stdio (if done in parallel/async).

Thanks!

**Comment by dstreet26** [comment]
I also saw this issue when compiling on Windows. I added some logging and also saw the `[WinError 206]` error...

I was able to get past the error by simply moving the repo folder to a higher directory (I tested `C:\`).
